File: memory-garden-server/src/main/java/com/memorygarden/model/entity/StudyPack.java
package com.memorygarden.model.entity;

import lombok.Data;

import java.util.Date;

/**
 * 预设知识包实体
 *
 * @author jLU
 * @date 2026-05-20
 */
@Data
public class StudyPack {

    /**
     * 主键
     */
    private Long id;

    /**
     * 知识包名称
     */
    private String name;

    /**
     * 知识包描述
     */
    private String description;

    /**
     * 建议分类名称
     */
    private String categoryName;

    /**
     * 包含卡片数
     */
    private Integer cardCount;

    /**
     * 软删除：0-未删除，1-已删除
     */
    private Integer isDeleted;

    /**
     * 创建时间
     */
    private Date createTime;

    /**
     * 更新时间
     */
    private Date updateTime;
}

File: memory-garden-web/src/api/studyPack.ts
import { get, post } from './request'

/**
 * 预设知识包实体
 */
export interface StudyPack {
  id: number
  name: string
  description: string
  categoryName: string
  cardCount: number
  isDeleted: number
  createTime: string
  updateTime: string
}

/**
 * 知识包条目实体
 */
export interface StudyPackItem {
  id: number
  packId: number
  frontContent: string
  backContent: string
  note: string
  sortOrder: number
  isDeleted: number
  createTime: string
}

/**
 * 预设知识包 API 封装
 */
export const studyPackApi = {
  /**
   * 获取知识包列表
   */
  list() {
    return get<StudyPack[]>('/study-pack/list')
  },

  /**
   * 获取知识包详情
   */
  getDetail(id: number) {
    return get<StudyPack>(`/study-pack/${id}`)
  },

  /**
   * 获取知识包条目列表
   */
  getPackItems(id: number) {
    return get<StudyPackItem[]>(`/study-pack/${id}/items`)
  },

  /**
   * 导入知识包
   */
  importPack(id: number) {
    return post<number>(`/study-pack/${id}/import`)
  }
}
